core/src/main/java/com/linecorp/armeria/common/multipart/MimeParser.java

```diff
@@ -201,42 +202,30 @@ void parse() {
                             return;
                         }
                         logger.trace("Skipped the preamble.");
-                        state = State.START_PART;
+                        state = State.START_PART_OR_END_MESSAGE;
                         break;
 
-                    case START_PART:
-                        logger.trace("state={}", State.START_PART);
+                    case START_PART_OR_END_MESSAGE:
+                        logger.trace("state={}", State.START_PART_OR_END_MESSAGE);
+                        final String headerLine = readHeaderLine();
```

Comment:
As per https://www.w3.org/Protocols/rfc1341/7_2_Multipart.html, `\r\n` could be appended to the closing boundary.
```
--simple boundary--\r\nThis is the epilogue.  It is also to be ignored.
```
Could we also consider the corner case?

Reply:
Thanks! fixed. 😉